File: src/app/api/models/Absence.ts
import { Employe } from './employe';

export interface Absence {
  id?: number;
  type?: string;
  motif?:string;
  dateDebut?: string;
  dateFin?: string;
  idEmploye?:number;
  employe?: string;
  joursRestants?: number;
}

File: src/main/java/com/safer/RH/models/Absence.java
package com.safer.RH.models;


import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
public class Absence {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    private Long id;
    private String type;
    private String motif;
    private LocalDate dateDebut;
    private LocalDate dateFin;

    @ManyToOne
    private Employe employe;
}
